src/Components/TimePicker/Component.purs

```diff
@@ -13,46 +16,80 @@ import Halogen.HTML.Events as HE
 import Halogen.HTML.Properties as HP
 import Ocelot.Block.Input as Input
 import Ocelot.Block.Layout as Layout
-import Ocelot.Component.TimePicker.Utils as Utils
+import Ocelot.Components.TimePicker.Utils as Utils
 import Ocelot.Data.DateTime as ODT
 import Ocelot.HTML.Properties (css)
-import Select as Select
+import Select as S
 import Select.Setters as Setters
 import Web.Event.Event (preventDefault)
 import Web.UIEvent.KeyboardEvent (KeyboardEvent)
 import Web.UIEvent.KeyboardEvent as KE
 
-type State =
-  { selection :: Maybe Time
-  , search :: String
+--------
+-- Types
+
+type Slot = H.Slot Query Output
+
+type Component m = H.Component HH.HTML Query Input Output m
+type ComponentHTML m = H.ComponentHTML Action ChildSlots m
+type ComponentRender m = State -> ComponentHTML m
+type ComponentM m a = H.HalogenM State Action ChildSlots Output m a
+
+type StateRow =
+  ( selection :: Maybe Time
   , timeUnits :: Array TimeUnit
   , disabled :: Boolean
-  }
+  )
+type State = Record StateRow
 
 type Input =
   { selection :: Maybe Time
   , disabled :: Boolean
   }
 
+defaultInput :: Input
+defaultInput =
+  { selection: Nothing
+  , disabled: false
+  }
+
+data Action
+  = PassingOutput Output
+
+data EmbeddedAction
+  = Search String
+  | Key KeyboardEvent
+-- TODO | Receive CompositeInput
+-- NOTE internal actions, moved to Util functions
+-- | Synchronize
+-- NOTE deprecated
+-- | TriggerFocus
+
 data Query a
-  = HandleSelect (Select.Message Query TimeUnit) a
-  | TriggerFocus a
-  | Synchronize a
-  | GetSelection (Maybe Time -> a)
+  = GetSelection (Time -> a)
   | SetSelection (Maybe Time) a
-  | Key KeyboardEvent a
-  | Search String a
 
-data Message
+data Output
   = SelectionChanged (Maybe Time)
-  | VisibilityChanged Select.Visibility
+  | VisibilityChanged S.Visibility
   | Searched String
 
-type ParentHTML m
-  = H.ParentHTML Query ChildQuery Input m
+type ChildSlots =
+  ( select :: S.Slot Query EmbeddedChildSlots Output Unit
+  )
+_select = SProxy :: SProxy "select"
+
+type CompositeState = S.State StateRow
+type CompositeAction = S.Action EmbeddedAction
+type CompositeQuery = S.Query Query EmbeddedChildSlots
+type CompositeInput = S.Input StateRow
+type EmbeddedChildSlots = () -- NOTE no extension, use S.Slot'
```

Comment:
The select component doesn't have any child components. The comment is incorrect though, we can't use `S.Slot'` here because we are extending selects queries with some of our own. I fixed the comment